src/Exceptionless.Core/Configuration/QueueOptions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using Exceptionless.Core.Extensions;
+using Foundatio.Utility;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Exceptionless.Core.Configuration {
+    public class QueueOptions {
+        public string ConnectionString { get; internal set; }
+        public string Provider { get; internal set; }
+        public Dictionary<string, string> Data { get; internal set; }
+
+        public string Scope { get; internal set; }
+        public string ScopePrefix { get; internal set; }
+    }
+
+    public class ConfigureQueueOptions : IConfigureOptions<QueueOptions> {
+        private readonly IConfiguration _configuration;
+
+        public ConfigureQueueOptions(IConfiguration configuration) {
+            _configuration = configuration;
+        }
+
+        public void Configure(QueueOptions options) {
+            options.Scope = _configuration.GetValue<string>(nameof(options.Scope), String.Empty);
+            options.ScopePrefix = !String.IsNullOrEmpty(options.Scope) ? options.Scope + "-" : String.Empty;
+
+            string cs = _configuration.GetConnectionString("queue");
+            options.Data = cs.ParseConnectionString();
+            options.Provider = options.Data.GetString(nameof(options.Provider).ToLowerInvariant());
```

Comment:
I want to rely on the IConfiguration system to get these settings and for it to provide the conversion helpers to coerce a string value into whatever type we need. It already has all of that functionality. So what we need to do is to parse the queue connection string into a dictionary like we are doing, but then to add those key/value pairs to the IConfiguration values in a sub-section called queue. Does that make sense?

Reply:
yeah, that makes sense but where does that happen? This is in configure and I don't think we can add to configuration at that time. You'd basically have to say add provider and these Configure options function is called multiple times.

Reply:
I just made some changes for this, so if you do `provider=redis` it will pull `config.getconnectionstring(provider)` and then append all those option pairs.